src/DSManager.cpp

```diff
@@ -405,15 +419,163 @@ Subscriber* DSManager::removeSubscriber(
     return ret;
 }
 
+ReturnCode_t DSManager::deleteSubscriber(
+        DataReader* dr)
+{
+    fastdds::dds::Subscriber* sub = nullptr;
+    {
+        std::lock_guard<std::recursive_mutex> lock(management_mutex);
+        GUID_t participant_guid = dr->get_subscriber()->get_participant()->guid();
+        sub = entity_map[participant_guid].subscriber;
+    }
+    return sub->delete_datareader(dr);
```

Comment:
Are we sure that this method is always going to be called with a valid DataReader pointer. If the subscriber is not found in the map or is it not initialized this instruction will cause a core dumped.

Even though this repository is not meant for public use, it is nevertheless public, and this method is part of the public API, so I think we should check the validity of the incoming parameters and ensure the code is robust.

This  comment applies to the other delete methods below.